src/main/java/com/google/cloud/spanner/jdbc/JdbcTypeConverter.java

```diff
@@ -199,6 +236,52 @@ static Object convert(Object value, Type type, Class<?> targetType) throws SQLEx
         com.google.rpc.Code.INVALID_ARGUMENT);
   }
 
+  /**
+   * Converts the given value to the Java {@link Class} type. The targetType {@link Class} must be
+   * an array of {@link AbstractMessage}.
+   */
+  static Object convertArrayOfProtoMessage(Object value, Class<?> targetType) throws Exception {
+    Class<?> componentType = targetType.getComponentType();
+    if (AbstractMessage.class.isAssignableFrom(componentType)) {
+      byte[][] result = (byte[][]) ((JdbcArray) value).getArray();
+      Object obj = java.lang.reflect.Array.newInstance(componentType, result.length);
+      for (int i = 0; i < result.length; i++) {
+        if (result[i] != null) {
+          Method parseMethodParseFrom = componentType.getMethod("parseFrom", byte[].class);
```

Comment:
nit: move `componentType.getMethod(..)` to outside the loop

Reply:
Done